From b73b802282bf95d214c86ba943c5765ba6930bc1 Mon Sep 17 00:00:00 2001
From: Jim Somerville <Jim.Somerville@windriver.com>
Date: Mon, 21 Oct 2013 12:50:44 -0400
Subject: [PATCH] Fix flawed dynamic -ldb link test in configure

The test uses dbopen, but just ignores the fact
that this function may not exist in the db version
used.  This leads to the dynamic link test failing
and the configure script just making assumptions
about why and setting the need for -ldb incorrectly.

Signed-off-by: Jim Somerville <Jim.Somerville@windriver.com>

---
 configure.ac | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/configure.ac b/configure.ac
index d2f7feb..c7ae568 100644
--- a/configure.ac
+++ b/configure.ac
@@ -3235,6 +3235,16 @@ case "$host" in
 		;;
 esac
 
+if test "x$ac_cv_have_decl_dbopen" = "xyes"; then
+  dnl 1.85
+  SQUID_CHECK_DBOPEN_NEEDS_LIBDB
+else
+  # dbopen isn't there. So instead of running a compile/link test that
+  # uses it and is thus guaranteed to fail, we just assume that we will
+  # need to link in the db library, rather than fabricate some other
+  # dynamic compile/link test.
+  ac_cv_dbopen_libdb="yes"
+fi
 dnl System-specific library modifications
 dnl
 case "$host" in
